assets/src/edit-story/components/library/panes/media/local/hotlink/hotlinkModal.js

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { __, sprintf, translateToExclusiveList } from '@web-stories-wp/i18n';
+import { Input } from '@web-stories-wp/design-system';
+import { useState, useCallback } from 'react';
+import styled from 'styled-components';
+import {
+  getFileNameFromUrl,
+  getFirstFrameOfVideo,
+  getImageDimensions,
+  getVideoDimensions,
+} from '@web-stories-wp/media';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { useConfig } from '../../../../../../app';
+import Dialog from '../../../../../dialog';
+import useLibrary from '../../../../useLibrary';
+import {
+  useUploadVideoFrame,
+  getPosterName,
+} from '../../../../../../app/media/utils';
+import { withProtocol } from '../../../../../../utils/url';
+
+const InputWrapper = styled.div`
+  margin: 16px 4px;
+  width: 470px;
+  height: 100px;
+`;
+
+function HotlinkModal({ isOpen, onClose }) {
+  const { allowedFileTypes } = useConfig();
+  const { insertElement } = useLibrary((state) => ({
+    insertElement: state.actions.insertElement,
+  }));
+
+  const { uploadVideoPoster } = useUploadVideoFrame({});
+
+  const [errorMsg, setErrorMsg] = useState(false);
+
+  // @todo We're not really uploading anything here, so should we have a fixed list instead?
+  let description = __('No file types are currently supported.', 'web-stories');
+  if (allowedFileTypes.length) {
+    description = sprintf(
+      /* translators: %s is a list of allowed file extensions. */
+      __('You can insert %s.', 'web-stories'),
+      translateToExclusiveList(allowedFileTypes)
+    );
+  }
+  const error = sprintf(
+    /* translators: %s is the description with allowed file extensions. */
+    __('Invalid link. %s', 'web-stories'),
```

Comment:
@swissspidy @o-fernandez Thoughts on the copy improvements here?

Currently looks like this:
![Screenshot 2021-08-10 at 12 22 35](https://user-images.githubusercontent.com/3294597/128842567-1a3ef42a-9e4f-41ca-9c29-9fa71b20e552.png)

Both the title (Currently just "Insert link") and the error message could probably be improved.

Additionally, there can't be linked to our editor due to CORS issues, this means that the media will fail to load in the editor, we also need an error message for this.

Could be something like "Media failed to load. Please ensure the link is valid and the site allows linking from external sites". Although this might not be a very helpful message.

Thoughts?

Reply:
Yeah "Insert link" sounds like I am adding a link to an element, not inserting media. Perhaps use "Insert external image or video"?

Then the input field should have a corresponding "URL" label for accessibility.

I think we'd also want some helper text at the top explaining what this feature does etc. So probably best to create a new ticket for refining that.

Something like "Media failed to load. Please ensure the link is valid and the site allows linking from external sites" is a good start I suppose. In a ticket, we can then figure out if there's a good FAQ resource for that, or an opportunity to link to the forums, etc.